it/v1/endpoints/IgnoreEmploymentExpensesControllerISpec.scala

```diff
@@ -0,0 +1,206 @@
+
+        def parserErrorTest(newNino: String, newTaxYear: String, newBody: JsValue, expectedStatus: Int, expectedBody: MtdError): Unit = {
+          s"parser returns ${expectedBody.code}" in new Test {
+
+            override val nino: String = newNino
+            override val taxYear: String = newTaxYear
+            override val requestBody: JsValue = newBody
+
+            override def setupStubs(): StubMapping = {
+              AuditStub.audit()
+              AuthStub.authorised()
+              MtdIdLookupStub.ninoFound(nino)
+            }
+
+            val response: WSResponse = await(request().put(requestBody))
+            response.status shouldBe expectedStatus
+            response.json shouldBe Json.toJson(expectedBody)
+          }
+        }
+
+        def currentYear: String = {
+          val currentDate = LocalDate.now()
+
+          val taxYear: Int = currentDate.getYear + 1
+
+          DesTaxYear.fromDesIntToString(taxYear)
+        }
+
+        val input = Seq(
+          ("AA123456ABCDEF", "2019-20", Json.parse(s"""{"ignoreExpenses": true}"""), BAD_REQUEST, NinoFormatError),
+          ("AA123456A", "201920", Json.parse(s"""{"ignoreExpenses": true}"""), BAD_REQUEST, TaxYearFormatError),
+          ("AA123456A", "2016-17", Json.parse(s"""{"ignoreExpenses": true}"""), BAD_REQUEST, RuleTaxYearNotSupportedError),
+          ("AA123456A", "2019-21", Json.parse(s"""{"ignoreExpenses": true}"""), BAD_REQUEST, RuleTaxYearRangeInvalidError),
+          ("AA123456A", currentYear, Json.parse(s"""{"ignoreExpenses": true}"""), FORBIDDEN, RuleTaxYearNotEndedError),
+          ("AA123456A", "2019-20", Json.parse(s"""{}"""), BAD_REQUEST, RuleIncorrectOrEmptyBodyError),
+        )
+
+        input.foreach(args => (parserErrorTest _).tupled(args))
+
+      }
+
+      "des service error" when {
+        def serviceErrorTest(desStatus: Int, desCode: String, expectedStatus: Int, expectedBody: MtdError): Unit = {
+          s"des returns an $desCode error and status $desStatus" in new Test {
+
+            override def setupStubs(): StubMapping = {
+              AuditStub.audit()
+              AuthStub.authorised()
+              MtdIdLookupStub.ninoFound(nino)
+              DesStub.onError(DesStub.PUT, desUri, desStatus, errorBody(desCode))
+            }
+
+            val response: WSResponse = await(request().put(requestBody))
+            response.status shouldBe expectedStatus
+            response.json shouldBe Json.toJson(expectedBody)
+          }
+        }
+
+        val input = Seq(
+          (BAD_REQUEST, "INVALID_TAXABLE_ENTITY_ID", BAD_REQUEST, NinoFormatError),
+          (BAD_REQUEST, "INVALID_TAX_YEAR", NOT_FOUND, NotFoundError),
+          (BAD_REQUEST, "INVALID_CORRELATIONID", INTERNAL_SERVER_ERROR, DownstreamError),
+          (BAD_REQUEST, "INVALID_PAYLOAD", INTERNAL_SERVER_ERROR, DownstreamError),
+          (UNPROCESSABLE_ENTITY, "INVALID_REQUEST_BEFORE_TAX_YEAR_END", FORBIDDEN, RuleTaxYearNotEndedError),
```

Comment:
Just a comment, seems a little odd that on the ignore endpoint RULE_TAX_YEAR_NOT_ENDED is status 403 while on amend it is 400

Reply:
Good point, I'll chase this up.

Reply:
You're right, it should've been 400.